cip/1.accepted/CIP2021-08-10-Operator-precedence.adoc

```diff
@@ -0,0 +1,725 @@
+Closed unary operators that allow chaining on the same precedence level are
+
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=UnaryAddOrSubtractExpression[Arithmetic additive inverse]
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=NotExpression[Boolean negation]
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=ListOperatorExpression[List slicing]
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=NullOperatorExpression[Null predicates]
+
+Example expressions are
+
+* `--+-5`
+* `NOT NOT NOT false`
+* `list[4..17][2..5]`
+* `x IS NOT NULL IS NOT NULL IS NULL`
+
+[IMPORTANT]
+.Chaining of closed unary operators
+====
+Chains of closed unary operators have an unambiguous operator tree.
+====
+
+[NOTE]
+.Associativity of unary operators
+====
+The grammar defines if the operand of an unary operator is on the left end or on the right end of the operator's syntax.
+This renders the operator left- or right associative, respectively.
+Nevertheless, the grammar allows only one unambiguous operator tree, in both case.
+====
+
+
+==== Chains of binary operators
+Binary operators that allow chaining with themselves of other operators of compatible result type on the same precedence level are
+
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=PowerOfExpression[Exponentiation]
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=MultiplyDivideModuloExpression[Multiplication]
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=MultiplyDivideModuloExpression[Division]
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=MultiplyDivideModuloExpression[Remainder (Modulo)]
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=AddOrSubtractExpression[Addition]
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=AddOrSubtractExpression[Substraction]
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=AndExpression[Boolean conjunction]
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=OrExpression[Boolean inclusive disjunction]
+* https://raw.githack.com/openCypher/openCypher/master/tools/grammar-production-links/grammarLink.html?p=OrExpression[Boolean exclusive disjunction]
+
+Example expressions are
+
+* `5 * 4 * 3`
+* `5 ^ 4 ^ 3`
+* `5 / 4 / 3`
+* `5 % 4 % 3`
+* `5 % 4 * 3`
+* `5 - 4 - 3`
+* `5 - 4 + 3`
+* `5 + 4 + 3`
+* `TRUE AND FALSE AND TRUE`
+* `TRUE OR FALSE OR TRUE`
+* `TRUE XOR FALSE XOR TRUE`
```

Comment:
The Cypher style guide states that: The null value and boolean literals should be written in lower case in a query.

```
* `true AND false AND true`
* `true OR false OR true`
* `true XOR false XOR true`
```